### PR TITLE
feat: implement withdraw

### PR DESCRIPTION
**Pull request by parketh**
## Summary

This PR implements the withdraw flow for op bridge by adding:
1. a new client-side IndexedDB instance to store and resume withdraws
2. a new `WithdrawTxStatus` component to show the progress of current withdraw tx

Once a withdraw is started the UI will not allow user to submit more withdraws until current one completes. However the DB can store multiple withdraws, including completed ones.

## Test plan

Run dev server to test

```
yarn dev
```

See example finalize withdraw tx: https://sepolia.etherscan.io/tx/0x372bd0d7fa7165de56a04ef2b02208f6ed66501eb15b776b41a6240a029ff348

image

### CODE REVIEW

**Comment by bap2pecs** [comment]
image

"prove" can be confusing for people who are not familiar with the OP Stack. can we add more description in this pop-up? e.g. to explain the withdrawal steps and what each step does

File: store/dexie/db.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  proveArgs?: any;
+  proveHash?: string;
+  proveReceipt?: any;
+  finalizeHash?: string;
+  finalizeReceipt?: any;
+  createdAt: Date;
+  updatedAt: Date;
+}
+
+export class MySubClassedDexie extends Dexie {
```
Comment (bap2pecs) [inline store/dexie/db.ts:29]: let's rename it to sth more meaningful like `WithdrawDexieDatabase`

**Comment by bap2pecs** [comment]
I tested the resume withdrawal feature. works pretty well:

image

**Review by bap2pecs — APPROVED** [review]
Take Off Fire - TRFcNpyLTdo2kUmkwm

File: store/withdraws.slice.ts
Comment (bap2pecs) [inline store/withdraws.slice.ts:1]: why is this removed?

**Review by bap2pecs — APPROVED** [review]

File: app/page.tsx [hunk 2]
```diff
@@ -568,24 +589,24 @@ export default function Bridge() {
...
       if (withdrawStatus === "proving") {
         await handleWithdrawAddProveReceipt();
       }
 
-      if (withdrawStatus === "proved") {
-        await handleWithdrawFinalize();
-      }
+      // if (withdrawStatus === "proved") {
+      //   await handleWithdrawFinalize();
+      // }
```
[the remark this answers is not in the document]
Comment (parketh) [reply app/page.tsx:602]: we no longer trigger these steps automatically. the user should click the button to manually trigger

File: store/withdraws.slice.ts
Comment (parketh) [reply store/withdraws.slice.ts:1]: this was a different way to implement the db which is no longer needed. we can just use the `get`, `set` and `update` api to modify the dexie db directly (its much simpler)

File: app/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bap2pecs) [reply app/page.tsx]: got it. if that's the case, let's remove the commented-out code
